addons/web_dashboard/models/res_users.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+
+from datetime import datetime, timedelta
+
+from openerp import models, api, _
+from openerp.exceptions import RedirectWarning
+from openerp.tools.misc import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class Users(models.Model):
+    _inherit = 'res.users'
+
+    def web_dashboard_get_users_by_state(self):
+        users = self.search([('login_date', '=', False)], order="create_date desc")
+        pending = users.filtered(lambda u: u.signup_valid)
+        expired = users - pending
+        return {'expired': zip(expired.mapped('id'), expired.mapped('login')), 'pending': zip(pending.mapped('id'), pending.mapped('login'))}
+
+    @api.model
+    def web_dashboard_create_users(self, email_adderesses, optional_message=""):
+        email_adderesses = map(str.strip, email_adderesses)
+        existing_users = self.search([('login', 'in', email_adderesses), '|', ('active', '=', True), ('active', '=', False)])
+        #Processing existing users
+        for user in existing_users:
+            if user.login_date == False and not user.signup_valid:
+                user.action_reset_password()
+            else:
+                #Already a user, reactivated in case it is deactivated
+                user.active = True
+        unused_emails = set(email_adderesses) - set(existing_users.mapped('login'))
+        #Processing unused email adresses : create new users (Used template user)
+        template_user_id = self.env['ir.config_parameter'].get_param('auth_signup.template_user_id', 'False')
+        for email in unused_emails:
+            default_values = {'login': email, 'name': email, 'email': email, 'active': True}
+            if template_user_id:
+                user = self.with_context(custom_message=optional_message, create_user=1).browse(int(template_user_id)).copy(default=default_values)
+            else:
+                user = self.with_context(custom_message=optional_message, create_user=1).create(default_values)
+            # if user haven't confugered Outgoing mail server
+            if user.signup_token == False:
```

Comment:
signup_token is False -> automatically linked to the outgoing mail server ? Are you sure ?